yt/frontends/swift/data_structures.py

```diff
@@ -97,7 +98,12 @@ def _parse_parameter_file(self):
         # Read from the HDF5 file, this gives us all the info we need. The rest
         # of this function is just parsing.
         header = self._get_info_attributes("Header")
-        runtime_parameters = self._get_info_attributes("RuntimePars")
+        # RuntimePars were removed from snapshots at SWIFT commit 6271388
+        with h5py.File(self.filename, mode="r") as handle:
+            has_runtime_pars = "RuntimePars" in handle.keys()
+
+        if has_runtime_pars:
+            runtime_parameters = self._get_info_attributes("RuntimePars")
```

Comment:
I'd rather we avoid possibly unbound names

```suggestion
        if has_runtime_pars:
            runtime_parameters = self._get_info_attributes("RuntimePars")
        else:
            runtime_parameters = {}
```

Reply:
Should we then also always add (the potentially empty) runtime_parameters to `self.parameters`?

Reply:
I guess it wouldn't hurt, especially if it simplifies the logic.